src/main/java/seedu/address/commons/util/StringUtil.java

```diff
@@ -38,6 +38,49 @@ public static boolean containsWordIgnoreCase(String sentence, String word) {
                 .anyMatch(preppedWord::equalsIgnoreCase);
     }
 
+    /**
+     * Returns true if the {@code sentence} contains any partial of the {@code word}.
+     *   Ignores case, and a partial word match is allowed, but should always match from
+     *   first letter of the word
+     *   <br>examples:<pre>
+     *       containsWordIgnoreCase("ABc def", "abc") == true
+     *       containsWordIgnoreCase("ABc def", "AB") == true
+     *       containsWordIgnoreCase("ABc def", "Bc") == false //not a full word match
+     *       </pre>
+     * @param sentence cannot be null
+     * @param word cannot be null, cannot be empty, must be a single word
+     */
+    public static boolean containsPartialWordIgnoreCase(String sentence, String word) {
+        requireNonNull(sentence);
+        requireNonNull(word);
+
+        String preppedWord = word.trim();
+        checkArgument(!preppedWord.isEmpty(), "Word parameter cannot be empty");
+        checkArgument(preppedWord.split("\\s+").length == 1, "Word parameter should be a single word");
+
+        String preppedSentence = sentence;
+        String[] wordsInPreppedSentence = preppedSentence.split("\\s+");
+
+        int len = word.length();
+
+        for (String preppedWords : wordsInPreppedSentence) {
+            int wordLength = preppedWords.length();
+            if (preppedWords.equalsIgnoreCase(word)) {
+                return true;
+            } else if (wordLength >= len) {
+                for (int i = 0; i < len; i++) {
+                    String preppedCharToString = preppedWords.substring(0, len);
+                    if (preppedCharToString.equalsIgnoreCase(word)) {
+                        return true;
+                    }
+                }
+            } else {
+                return false;
+            }
+        }
+        return false;
```

Comment:
I think preppedWords should be renamed to another name to avoid confusion with preppedWord @ line 57.

Additionally, for the if part, it can be shortened to:

`return (preppedWords.toLowerCase().contains(word.toLowerCase())`

Reply:
Thanks for the review! Yes I just realised the for loop is not used at all so I will be updating it to your suggestion